### PR TITLE
Bundle micstreamselector dll and updated MRTK scripts into an NPM package

### PR DESCRIPTION
**Pull request by davidkline-ms**

### CODE REVIEW

File: Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+            if (!initialized)
+            {
+                // The microphone stream is not initialized, no need to alarm the calling code.
+                return WindowsMicrophoneStreamErrorCode.Success;
+            }
+
+            return (WindowsMicrophoneStreamErrorCode)MicDestroy();
+        }
+
+#region MicStream.dll methods
```
Comment (keveleigh) [inline Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs:276]: nit: some inconsistency on the `#` indentation (some regions and ifs are indented, some aren't). I think typically we default to not indenting ifs and indenting regions. VS's formatter should do that by default

File: Input/MicStreamSelector/Solutions/MicStreamSelector.sln [hunk 2]
```diff
@@ -1,79 +1,67 @@
...
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Debug|x86.Deploy.0 = Debug|x86
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|ARM.ActiveCfg = Release|ARM
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|ARM.Build.0 = Release|ARM
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|ARM.Deploy.0 = Release|ARM
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|ARM64.ActiveCfg = Release|x86
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|x64.ActiveCfg = Release|x64
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|x64.Build.0 = Release|x64
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|x64.Deploy.0 = Release|x64
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|x86.ActiveCfg = Release|x86
-		{E1E23D23-2A2E-4E03-8C2F-112D27365BB9}.Release|x86.Build.0 = Release|x86
```
Comment (keveleigh) [inline Input/MicStreamSelector/Solutions/MicStreamSelector.sln:42]: Is this turning off specific builds? Release|x86 seems like a supported configuration

**Review by keveleigh — DISMISSED** [review]

File: Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj [hunk 3]
```diff
@@ -1,276 +1,276 @@
...
+    <GenerateManifest>false</GenerateManifest>
+    <IgnoreImportLibrary>false</IgnoreImportLibrary>
+    <TargetName>MicStreamSelector</TargetName>
+    <OutDir>$(SolutionDir)$(Platform)\$(Configuration)\$(MSBuildProjectName)\</OutDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <GenerateManifest>false</GenerateManifest>
+    <IgnoreImportLibrary>false</IgnoreImportLibrary>
+    <TargetName>MicStreamSelector</TargetName>
+    <OutDir>$(SolutionDir)$(Platform)\$(Configuration)\$(MSBuildProjectName)\</OutDir>
```
Comment (keveleigh) [inline Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj:182]: I'm not sure these `OutDir`s match the expectation of the packaging scripts. The UnityAddon folder has a "UWP" folder, but that's not present here. Does it still work?

Comment (davidkline-ms) [reply Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj:182]: i copied this from plane finding.... i'll update :)

File: Input/MicStreamSelector/Solutions/MicStreamSelector.sln (hunk 2 above)
Comment (davidkline-ms) [reply Input/MicStreamSelector/Solutions/MicStreamSelector.sln:42]: i'll restore these. was an attempt to get the build working

File: Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs (hunk 1 above)
Comment (davidkline-ms) [reply Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs:276]: that may be from the legacy code. happy to fix 😄

File: Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj (hunk 3 above)
Comment (keveleigh) [reply Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj:182]: Plane finding has the "UWP" path: https://github.com/microsoft/MixedRealityToolkit/blob/6f5b6121cf44a952e5306c2b5697b19c9ac98095/SpatialMapping/PlaneFinding/PlaneFinding/PlaneFinding.UWP/PlaneFinding.UWP.vcxproj#L137

It's different across the two vcxproj files (UWP and Win32)

File: Input/MicStreamSelector/Solutions/MicStreamSelector.sln [hunk 4]
```diff
@@ -12,19 +12,16 @@ EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|ARM = Debug|ARM
-		Debug|ARM64 = Debug|ARM64
```
Comment (keveleigh) [inline Input/MicStreamSelector/Solutions/MicStreamSelector.sln:15]: Removing ARM64 intentionally?

File: Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj (hunk 3 above)
Comment (davidkline-ms) [reply Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj:182]: i guess i just missed it 😄

File: Input/MicStreamSelector/Solutions/MicStreamSelector.sln (hunk 4 above)
Comment (davidkline-ms) [reply Input/MicStreamSelector/Solutions/MicStreamSelector.sln:15]: no, that mustve happened when i restored the sln from the primary branch. weird

Comment (davidkline-ms) [reply Input/MicStreamSelector/Solutions/MicStreamSelector.sln:15]: i do not recall modifying that line. can you double check your clone?

Comment (keveleigh) [reply Input/MicStreamSelector/Solutions/MicStreamSelector.sln]: this looks resolved

File: Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+        [DllImport("MicStreamSelector", ExactSpelling = true)]
+        private static extern int MicStartStream(bool keepData, bool previewOnDevice, LiveMicCallback micsignal = null);
+
+        [DllImport("MicStreamSelector", ExactSpelling = true)]
+        private static extern void MicStopRecording(StringBuilder sb);
+
+        [DllImport("MicStreamSelector", ExactSpelling = true)]
+        private static extern int MicStopStream();
+
+    #endregion MicStream.dll methods
```
Comment (keveleigh) [inline Input/MicStreamSelector/UnityAddon/WindowsMicrophoneStream.cs:311]:
```suggestion
        #endregion MicStream.dll methods
```

File: Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj (hunk 3 above)
Comment (keveleigh) [reply Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj:182]: I don't think these paths have been updated yet

Comment (davidkline-ms) [reply Input/MicStreamSelector/Projects/MicStreamSelector_UWP/MicStreamSelector_UWP.vcxproj]: Believe i have addressed it this time 😄

**Review by keveleigh — APPROVED** [review]
Approving to unblock check-in

**Pull request merged**